orchagent/fdborch.cpp

```diff
@@ -63,3 +70,218 @@ bool FdbOrch::getPort(const MacAddress& mac, uint16_t vlan, Port& port)
+                it = consumer.m_toSync.erase(it);
+                continue;
+            }
+
+            // check that type either static or dynamic
+            if (type != "static" && type != "dynamic")
+            {
+                SWSS_LOG_ERROR("FDB entry with key: '%s' has type '%s'. But allowed only types: 'static' or 'dynamic'\n",
+                               key.c_str(), type.c_str());
+                it = consumer.m_toSync.erase(it);
+                continue;
+            }
+
+            if (addFdbEntry(entry, port, type))
+                it = consumer.m_toSync.erase(it);
+            else
+                it++;
+        }
+        else if (op == DEL_COMMAND)
+        {
+            if (removeFdbEntry(entry))
+                it = consumer.m_toSync.erase(it);
+            else
+                it++;
+
+        }
+        else
+        {
+            SWSS_LOG_ERROR("Unknown operation type %s\n", op.c_str());
+            it = consumer.m_toSync.erase(it);
+        }
+    }
+}
+
+bool FdbOrch::addFdbEntry(const FdbEntry& entry, const string& port_name, const string& type)
+{
+    SWSS_LOG_ENTER();
+
+    SWSS_LOG_DEBUG("mac=%s, vlan=%d. port_name %s. type %s\n",
+                   entry.mac.to_string().c_str(), entry.vlan, port_name.c_str(), type.c_str());
+
+    if (m_entries.count(entry) != 0) // we already have such entries
+    {
+        // FIXME: should we check that the entry are moving to another port?
+        // FIXME: should we check that the entry are changing its type?
+        SWSS_LOG_ERROR("FDB entry already exists. mac=%s vlan=%d\n", entry.mac.to_string().c_str(), entry.vlan);
+        return true;
+    }
+
+    sai_status_t status;
+
+    sai_fdb_entry_t fdb_entry;
+    memcpy(fdb_entry.mac_address, entry.mac.getMac(), sizeof(sai_mac_t));
+    fdb_entry.vlan_id = entry.vlan;
+
+    Port port;
+    if (!m_portsOrch->getPort(port_name, port))
+    {
+        SWSS_LOG_ERROR("Failed to get port id for %s\n", port_name.c_str());
+        return true;
+    }
+
+    sai_attribute_t attr;
+    vector<sai_attribute_t> attrs;
+
+    attr.id = SAI_FDB_ENTRY_ATTR_TYPE;
+    attr.value.u8 = (type == "dynamic") ? SAI_FDB_ENTRY_DYNAMIC : SAI_FDB_ENTRY_STATIC; // FIXME: u8 is right here?
+    attrs.push_back(attr);
+
+    attr.id = SAI_FDB_ENTRY_ATTR_PORT_ID;
+    attr.value.oid = port.m_port_id;
+    attrs.push_back(attr);
+
+    attr.id = SAI_FDB_ENTRY_ATTR_PACKET_ACTION;
+    attr.value.u8 = SAI_PACKET_ACTION_FORWARD; // FIXME: what to use here? what about u8 type?
+    attrs.push_back(attr);
+
+    status = sai_fdb_api->create_fdb_entry(&fdb_entry, attrs.size(), attrs.data());
+    if (status != SAI_STATUS_SUCCESS)
+    {
+        SWSS_LOG_ERROR("Failed to add FDB entry. mac=%s, vlan=%d. port_name %s. type %s\n",
+                   entry.mac.to_string().c_str(), entry.vlan, port_name.c_str(), type.c_str());
+        return false; // FIXME: retry it? How many times?
```

Comment:
no need to retry, currently the status is returned by sairedis, which will always be success as long as you pass the parameter validation.